modules/alloy/src/main/ts/ephox/alloy/positioning/layout/LayoutBounds.ts

```diff
@@ -0,0 +1,22 @@
+import * as Boxes from '../../alien/Boxes';
+import { BubbleInstance } from './Bubble';
+import { AnchorBox } from './LayoutTypes';
+
+type BoundFunc = (bounds: number, anchor: AnchorBox, bubbleOffset: number) => number;
+
+export const anchorLeft = (bounds: number, anchor: AnchorBox, bubbleOffset: number) => anchor.x() + bubbleOffset;
+export const anchorRight = (bounds: number, anchor: AnchorBox, bubbleOffset: number) => anchor.x() + anchor.width() + bubbleOffset;
+export const anchorTop = (bounds: number, anchor: AnchorBox, bubbleOffset: number) => anchor.y() + bubbleOffset;
+export const anchorBottom = (bounds: number, anchor: AnchorBox, bubbleOffset: number) => anchor.y() + anchor.height() + bubbleOffset;
+
+export const adjustBounds = (bounds: Boxes.Bounds, anchor: AnchorBox, bubble: BubbleInstance, left: BoundFunc, top: BoundFunc, right: BoundFunc, bottom: BoundFunc) => {
```

Comment:
It's probably a lot clearer in an IDE, but I wonder if we should group together `left`, `top` etc. into an object so that it's one argument.

Reply:
This is the same order as the `Boxes.bounds` so I want to keep it the same as that.